app/src/main/java/com/bogdan/codeforceswatcher/features/problems/ProblemsFragment.kt

```diff
@@ -1,126 +1,389 @@
-import kotlinx.android.synthetic.main.fragment_problems.*
 import tw.geothings.rekotlin.StoreSubscriber
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.BasicTextField
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.Icon
+import androidx.compose.material.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.SolidColor
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.unit.IntSize
+import androidx.compose.ui.unit.dp
+import com.bogdan.codeforceswatcher.components.WebViewActivity
+import io.xorum.codeforceswatcher.util.AnalyticsEvents
+import com.bogdan.codeforceswatcher.R
+import com.google.accompanist.swiperefresh.SwipeRefresh
+import com.google.accompanist.swiperefresh.SwipeRefreshState
+import com.google.accompanist.swiperefresh.rememberSwipeRefreshState
+import io.xorum.codeforceswatcher.features.problems.redux.ProblemsRequests
+import io.xorum.codeforceswatcher.redux.analyticsController
 
-class ProblemsFragment : Fragment(), StoreSubscriber<ProblemsState>,
-    SwipeRefreshLayout.OnRefreshListener {
-
-    private lateinit var problemsAdapter: ProblemsAdapter
-    var searchView: SearchView? = null
+class ProblemsFragment : Fragment(), StoreSubscriber<ProblemsState> {
 
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        setHasOptionsMenu(true)
-    }
+    private val problemsState: MutableState<List<UIModel>> = mutableStateOf(emptyList())
+    private val isRefreshState: SwipeRefreshState = SwipeRefreshState(false)
 
     override fun onCreateView(
-        inflater: LayoutInflater,
-        container: ViewGroup?,
+        inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View = inflater.inflate(R.layout.fragment_problems, container, false)
-
-    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        super.onViewCreated(view, savedInstanceState)
-        initViews()
+    ) = ComposeView(requireContext()).apply {
+        setContent {
+            AlgoismeTheme {
+                ProblemsView(
+                    problemsState = problemsState,
+                    isRefreshState = isRefreshState,
+                    onProblem = { link, title ->
```

Comment:
Nit: can we one-line `onProblem` and `onStar`as well?

Reply:
But these are lambdas?, the parameters and the followed code should be separated by a new line.

Reply:
Not always, you can write them on a single line as well.